draft-ietf-taps-interface.md

```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
 on receiving them.
 
 
+### Active side is not the first to write {#active-read-before-send}
+
+Name:
+: activeReadBeforeSend
+
+Type:
+: Preference
+
+Default:
+: Ignore
+
+The most common client-server communication pattern involves the
+client actively opening a connection, then sending data to the server. The
+server listens (passive open), reads, and then answers. This property
+specifies whether an application wants to diverge from this pattern -- either
+by actively opening, immediately followed by reading, or passively opening,
+immediately followed by writing.
```

Comment:
I get the problem for read-after-active-open, but what is the problem with passive open and immediately writing – this is fairly common in TCP based protocols and you can easily put the write in the connection ready.

Reply:
It's the same thing with multi-streaming: a stream only comes into being when the first message is sent on it. So, "passive open and immediately writing" doesn't map onto anything with streams - there is no such thing as a stream becoming available without data arriving on it.